libraries/lib-datahandler/src/main/thrift/vulnerabilities.thrift

```diff
@@ -218,4 +234,16 @@ service VulnerabilityService {
       * returns null otherwise
       **/
     Vulnerability getVulnerabilityByExternalId(1: string externalId, 2: User user);
+
+    /**
+      * returns the vulnerability with given externalId if it exists in the database and if user is valid
+      * returns null otherwise
+      **/
+    list<Vulnerability> getVulnerabilitiesByExternalIdOrConfiguration(1: optional string externalId, 2: optional string vulnerableConfiguration, 3: User user);
```

Comment:
`optional` arguments in a function call? I do not think that we use that somewhere else.

Reply:
We do not, indeed, but since they are not required together, why not?